web/search.jsp

```diff
@@ -65,11 +65,18 @@ include file="projects.jspf"
     PageConfig cfg = PageConfig.get(request);
 
     long starttime = System.currentTimeMillis();
-
-    SearchHelper searchHelper = cfg.prepareSearch();
+	SearchHelper searchHelper = null;
+    if (request.getParameter("hdnSearchAll") != null && request.getParameter("hdnSearchAll").equals("true"))
```

Comment:
`request.getParameter("hdnSearchAll") != null && request.getParameter("hdnSearchAll").equals("true")`

I would suggest to make a method of PageConfig to check this, something like `isSearchAll` or similar.